python/kvikio/cufile.py

```diff
@@ -260,8 +280,76 @@ def write(
         """
         return self.pwrite(buf, size, file_offset, task_size).get()
 
+    def raw_read_async(
+        self,
+        buf,
+        stream,
+        size: Optional[int] = None,
+        file_offset: int = 0,
+        dev_offset: int = 0,
+    ) -> IOFutureStream:
+        """Reads specified bytes from the file into the device memory asynchronously
+
+        This is an async version of `.raw_read` that doesn't use threads and
+        does not support host memory.
+
+        Parameters
+        ----------
+        buf: buffer-like or array-like
+            Device buffer to read into.
+        size: int, optional
+            Size in bytes to read.
+        file_offset: int, optional
+            Offset in the file to read from.
+        stream: cuda.Stream, optional
+            CUDA stream to perform the read operation asynchronously.
+
+        Returns
+        -------
+        IOFuture
+            Future that when executed ".check_bytes_done()" returns the size of bytes
+            that were successfully read.
+        """
+        return self._handle.read_async(buf, size, file_offset, dev_offset, stream)
+
+    def raw_write_async(
+        self,
+        buf,
+        stream,
+        size: Optional[int] = None,
+        file_offset: int = 0,
+        dev_offset: int = 0,
+    ) -> IOFutureStream:
+        """Writes specified bytes from the device memory into the file asynchronously
+
+        This is an async version of `.raw_write` that doesn't use threads and
+        does not support host memory.
+
+        Parameters
+        ----------
+        buf: buffer-like or array-like
+            Device buffer to write to.
+        size: int, optional
+            Size in bytes to write.
+        file_offset: int, optional
+            Offset in the file to write from.
+        stream: cuda.Stream, optional
+            CUDA stream to perform the write operation asynchronously.
+
+        Returns
+        -------
+        IOFuture
```

Comment:
```suggestion
        IOFutureStream
```

Also, add something like:
```
The instance must be kept alive alive until all data has been written to disk. One
    way to do this, is by calling `StreamFuture.check_bytes_done()`, which will
    synchronize the associated stream and return the number of bytes written."""
```